src/basilisp/contrib/pytest/testrunner.py

```diff
@@ -135,18 +135,43 @@ def teardown(self) -> None:
         self._teardowns = ()
 
 
+def _is_package(path: Path) -> bool:
+    """Return `True` if the given path refers to a Python or Basilisp package."""
+    _, _, files = next(os.walk(path))
+    for file in files:
+        if file in {"__init__.lpy", "__init__.py"} or file.endswith(".lpy"):
+            return True
+    return False
+
+
+def _get_fully_qualified_module_name(file: Path) -> str:
+    """Return the fully qualified module name (from the import root) for a module given
+    its location.
+
+    This works by traversing up the filesystem looking for the top-most package. From
+    there, we derive a Python module name referring to the given module path."""
+    top = None
+    for p in file.parents:
+        if _is_package(p):
+            top = p
+        else:
+            break
+
+    if top is None or top == file.parent:
+        return file.stem
+
+    root = top.parent
+    elems = list(file.with_suffix("").relative_to(root).parts)
+    if elems[-1] == "__init__":
+        elems.pop()
+    return ".".join(elems)
+
+
 class BasilispFile(pytest.File):
     """Files represent a test module in Python or a test namespace in Basilisp."""
 
-    def __init__(  # pylint: disable=too-many-arguments
-        self,
-        fspath: py.path.local,
-        parent=None,
-        config: Optional[Config] = None,
-        session: Optional["Session"] = None,
-        nodeid: Optional[str] = None,
-    ) -> None:
-        super().__init__(fspath, parent, config, session, nodeid)
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
```

Comment:
https://docs.pytest.org/en/stable/deprecations.html#constructors-of-custom-pytest-node-subclasses-should-take-kwargs